docs/user-guide/logging/overview.md

```diff
@@ -0,0 +1,88 @@
+
+In the following sections firstly a basic demo of working with application logs is given. Then the node logging architecture is described in details. After that, several solutions for cluster-level logging are laid out. If you're not interested in having cluster-level logging, you might still find the description how logs are stored and handled on the node useful. For the latter section the assumption is that there exists a logging backend, inside or outside of kubernetes cluster.
+
+## Logging demo
+
+For the sake of demonstration [a following pod specification](/docs/user-guide/logging/counter-pod.yaml) is used. It has a container which writes out some text to standard output every second.
+
+{% include code.html language="yaml" file="counter-pod.yaml" %}
+
+Pod can run using the following command
+
+```shell
+$ kubectl create -f ./counter-pod.yaml
+pod "counter" created
+```
+
+and then fetch the logs:
+
+```shell
+$ kubectl logs counter
+0: Tue Jun  2 21:37:31 UTC 2015
+1: Tue Jun  2 21:37:32 UTC 2015
+2: Tue Jun  2 21:37:33 UTC 2015
+3: Tue Jun  2 21:37:34 UTC 2015
+4: Tue Jun  2 21:37:35 UTC 2015
+5: Tue Jun  2 21:37:36 UTC 2015
+...
+```
+
+It's also possible to get logs from the previous instantiation of the container in case it crashed or specity the container if pods contains several of them. More details on the [`kubectl logs` page](/docs/user-guide/kubectl/kubectl_logs).
+
+## Logging one the node level
+
+![Node level logging](/images/docs/user-guide/logging/logging-node-level.png)
+
+Everything written to `stdout` and `stderr` by a containerized application goes to a container runtime. Currently, from the Docker runtime it goes to a file using standard [json logging driver](https://docs.docker.com/engine/admin/logging/overview). __Note! It implies that each _line_ will be treated as a separate message, no direct support for multiline messages, e.g. stacktraces.__ In case container restarts, kubelet keeps by default one terminated container with its logs. In case pod is evicted from the node, all corresponding containers are evicted too with their logs.
+
+Important question is how to achieve log rotations so that logs won't consume all available space on the node. Right now, logrotate tool solves this task. Detailed configuration can be found [there](https://github.com/kubernetes/kubernetes/blob/release-1.5/cluster/gce/gci/configure-helper.sh#L96). In short, up to `5` rotation are kept, rotation is performed daily or if log file grows beyond `10MB`. Rotations belong to a single container, that is if container dies several times or pod gets evicted, all rotations for the container are lost.
```

Comment:
"An important consideration in node-level logging is implementing log rotation, so that logs don't consume all available storage on the node. Kubernetes uses the `logrotate` tool to implement log rotation.

The default 'logrotate` behavior is to keep up to five logging rotations. Kubernetes performs log rotation daily, or if the log file grows beyond 10MB in size. Each rotation belongs to a single container; if the container repeatedly fails or the pod is evicted, all rotations for the container are lost.

You can find detailed information in the [configure helper](https://github.com/kubernetes/kubernetes/blob/release-1.5/cluster/gce/gci/configure-helper.sh#L96)."